File: fixtures/scale/dart/large/lib/ciphers/caesar/caesar_suite.dart
import '../../core/cipher.dart';
import '../../core/cipher_suite.dart';
import '../../core/suite_result.dart';
import '../../verify/round_trip.dart';
import 'caesar_cipher.dart';
import 'caesar_key.dart';
import 'caesar_vectors.dart';

class CaesarSuite implements CipherSuite {
  @override
  String name() => 'caesar';

  @override
  String category() => 'cipher';

  @override
  SuiteResult run() {
    final Cipher cipher = CaesarCipher(CaesarKey.defaultKey());
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in caesarSamples()) {
      if (roundTripCheck(cipher, sample)) {
        passed++;
      } else {
        failed++;
      }
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}

File: fixtures/scale/scala/large/src/cryptobench/ciphers/caesar/CaesarSuite.scala
package cryptobench.ciphers.caesar

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.RoundTrip

final class CaesarSuite extends CipherSuite {
  override def name: String = "caesar"

  override def run(): SuiteResult = {
    val cipher: Cipher = new CaesarCipher(CaesarKey.default())
    var passed = 0
    var failed = 0
    val start = System.nanoTime()
    for (sample <- CaesarVectors.samples()) {
      if (RoundTrip.check(cipher, sample)) passed += 1 else failed += 1
    }
    SuiteResult(name, passed, failed, System.nanoTime() - start)
  }
}
